activerecord/lib/active_record/connection_handling.rb

```diff
@@ -87,6 +90,23 @@ def connection
       retrieve_connection
     end
 
+    def specification_id=(value)
+      @specification_id = value
+    end
+
+    def specification_id(fallback = true)
+      return @specification_id if defined?(@specification_id)
+      find_legacy_spec_id(self) if fallback
+    end
+
+    def find_legacy_spec_id(klass)
+      return "primary" if klass == Base
+      if id = klass.specification_id(false)
+        return id
+      end
+      find_legacy_spec_id(klass.superclass)
+    end
```

Comment:
We need to do some gymnastics here to be backwards compatible.
However if we deprecate the `establish_connection` from the model, eventually we would be able to remove this fallback into the `Class.name` method.

The way I envision this, is we would establish the connection straight to the handler, as this connection is not attached to the model anymore.